zjit/src/backend/x86_64/mod.rs

```diff
@@ -183,7 +183,7 @@ impl Assembler
                                     *left = asm.load(*left);
                                 },
                                 // The first operand can't be an immediate value
```

Comment:
What does this comment mean? It looks like now it *can* be an immediate? Or maybe I am totally misreading

Reply:
There's no encoding where LHS is an immediate. This code turns such an immediate into a non-immediate by loading the immediate into a register.